impl/src/test/java/org/ehcache/spi/event/DefaultCacheEventListenerFactoryTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.spi.event;
+
+import org.ehcache.CacheManager;
+import org.ehcache.CacheManagerBuilder;
+import org.ehcache.config.CacheConfigurationBuilder;
+import org.ehcache.config.event.DefaultCacheEventListenerConfiguration;
+import org.ehcache.event.CacheEvent;
+import org.ehcache.event.CacheEventListener;
+import org.ehcache.event.EventFiring;
+import org.ehcache.event.EventOrdering;
+import org.ehcache.event.EventType;
+import org.junit.Test;
+
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Created by rism on 3/3/15.
+ */
+public class DefaultCacheEventListenerFactoryTest {
+
+  @Test
+  public void testCacheConfigUsage() {
+    DefaultCacheEventListenerConfiguration cacheEventListenerConfiguration = new DefaultCacheEventListenerConfiguration(ListenerObject.class);
```

Comment:
Why not have a builder for a listener configuration? Also, calling into a test object instead of using the enums directly makes the code less clear.